src/integer/mat_poly_over_z/coefficient_embedding.rs

```diff
@@ -0,0 +1,169 @@
+    /// the coefficients can not be embedded correctly.
+    /// - if `self` is not a row vector
+    fn into_coefficient_embedding(&self, size: impl Into<i64>) -> MatZ {
+        assert!(self.is_row_vector());
+        let size = size.into();
+
+        let mut poly: Vec<PolyOverZ> = Vec::new();
+        for i in 0..self.get_num_columns() {
+            let entry = self.get_entry(0, i).unwrap();
+            assert!(entry.get_degree() < size);
+            poly.push(entry);
+        }
+        let mut out = MatZ::new(size, poly.len());
+
+        for (i, entry) in poly.iter().enumerate() {
+            for j in 0..size {
+                match entry.get_coeff(j) {
+                    Ok(value) => out.set_entry(j, i, &value).unwrap(),
+                    Err(_) => break,
+                }
+            }
+        }
+
+        out
+    }
+}
+
+impl FromCoefficientEmbedding<&MatZ> for MatPolyOverZ {
+    /// Computes a row vector of polynomials from a matrix.
+    /// The j-th entry of the i-th column vector is taken
+    /// as the coefficient of the polynomial of the i-th polynomial.
+    ///
+    /// Parameters:
+    /// - `embedding`: the column vector that encodes the embedding
+    ///
+    /// Returns a row vector of polynomials that corresponds to the embedding
+    ///
+    /// # Examples
+    /// ```
+    /// use std::str::FromStr;
+    /// use qfall_math::{
+    ///     integer::{MatZ, MatPolyOverZ},
+    ///     traits::FromCoefficientEmbedding,
+    /// };
+    ///
+    /// let matrix = MatZ::from_str("[[17, 1],[3, 2],[-5, 3]]").unwrap();
+    /// let poly = MatPolyOverZ::from_coefficient_embedding(&matrix);
+    /// let cmp_poly = MatPolyOverZ::from_str("[[3  17 3 -5, 3  1 2 3]]").unwrap();
+    /// assert_eq!(cmp_poly, poly)
+    /// ```
+    fn from_coefficient_embedding(embedding: &MatZ) -> Self {
+        let mut out = MatPolyOverZ::new(1, embedding.get_num_columns());
+
+        for i in 0..embedding.get_num_columns() {
+            let col_vec = embedding.get_column(i).unwrap();
+            let entry = PolyOverZ::from_coefficient_embedding(&col_vec);
+            out.set_entry(0, i, entry).unwrap();
+        }
+        out
+    }
+}
+
+#[cfg(test)]
+mod test_into_coefficient_embedding {
+    use crate::{
+        integer::{MatPolyOverZ, MatZ},
+        traits::IntoCoefficientEmbedding,
+    };
+    use std::str::FromStr;
+
+    /// Ensure that the initialization of the identity matrix works
+    #[test]
+    fn standard_basis() {
+        let standard_basis = MatPolyOverZ::from_str("[[1  1, 2  0 1]]").unwrap();
+        let basis = standard_basis.into_coefficient_embedding(3);
+
+        assert_eq!(MatZ::identity(3, 2), basis)
+    }
+
+    /// Ensure that the embedding works with large entries
+    #[test]
+    fn large_entries() {
+        let cmp_poly =
+            MatPolyOverZ::from_str(&format!("[[3  17 {} {}, 1  1]]", i64::MAX, i64::MIN)).unwrap();
+        let matrix = cmp_poly.into_coefficient_embedding(3);
```

Comment:
```suggestion
        let poly =
            MatPolyOverZ::from_str(&format!("[[3  17 {} {}, 1  1]]", i64::MAX, i64::MIN)).unwrap();
            
        let matrix = poly.into_coefficient_embedding(3);
```
rename cmp_poly to poly, 
Add newline